sympy/integrals/transforms.py

```diff
@@ -1935,19 +2060,125 @@ def laplace_transform(f, t, s, legacy_matrix=True, **hints):
             else:
                 return type(f)(*f.shape, elements_trans)
 
-    return LaplaceTransform(f, t, s).doit(**hints)
+    # First remove a common factor Heaviside(t) if there is one.
+    g = f.collect(Heaviside(t))
+    if g.is_Mul:
+        factors = g.args
+        if Heaviside(t) in factors:
+            f = Mul(*[ i for i in factors if not i is Heaviside(t) ])
+
+    debug('[LT l_t  ]   executes LaplaceTransform(%s, %s, %s)'%(f, t, s))
+
+    LT = LaplaceTransform(f, t, s).doit(**hints)
+
+    debug('[LT l_t  ]   received: ', LT)
+
+    # For backwards compatibility: even if noconds=False, do not return
+    # a tuple if F is unevaluated
+    F, p, c = LT
+    if _laplace_is_uneval(F) or not conds:
+        LT = F
+    if F.has(LaplaceTransform) and p is S.NegativeInfinity and c:
+        LT = F
+
+    debug('[LT l_t  ]   returns: ', LT)
+    return LT
+
+def _complete_the_square_in_denom(f, s):
+    from sympy.simplify.radsimp import fraction
+    [n, d] = fraction(f)
+    if d.is_polynomial(s):
+        cf = d.as_poly(s).all_coeffs()
+        if len(cf)==3:
+            a, b, c = cf
+            d = a*((s+b/(2*a))**2+c/a-(b/(2*a))**2)
+    return n/d
+
+def _inverse_laplace_build_rules(s, t):
```

Comment:
I would suggest a small modification to this, which is to store the list at the module level and have this function populate the list on the first call. This would require replacing `s` and `t` with Dummy variables and replacing with the actual variables when the rule is applied (or alternately, use a lambda for each rule). 

That way, advanced users will be able to inject their own rules into the list to enable Laplace transforms for things that otherwise don't work.

Reply:
I like it but don't know how to do it. Is such a thing done in another place in the sympy code base, or could you give me a code fragment (does not need to be tested)? The thing I don't know is how to define dummies on the module level and replace them in the function.

Reply:
I don't like that idea because it then commits to making what should really be an internal data structure public so that it can't be changed in future. A better idea would be to have a function that users could call and pass in their own rules.

Reply:
Let's make a new issue for this, OK? I'd prefer to conclude this part of the work and merge it before adding functionality. The reason is that my health is not as robust as I would like, and it would be bad if I get stuck in a bigger project that I cannot finish in reasonable time.